sub-packages/bionemo-amplify/tests/bionemo/amplify/test_convert.py

```diff
@@ -0,0 +1,270 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from pathlib import Path
+
+import torch
+from megatron.core.transformer.module import Float16Module
+from nemo.lightning import io
+from transformers import AutoModel
+
+from bionemo.amplify.convert import HFAMPLIFYImporter  # noqa: F401
+from bionemo.amplify.hf_rotary import apply_rotary_emb
+from bionemo.amplify.model import AMPLIFYConfig
+from bionemo.amplify.tokenizer import BioNeMoAMPLIFYTokenizer
+from bionemo.core.utils.dtypes import PrecisionTypes, get_autocast_dtype
+from bionemo.esm2.testing.compare import ForwardHook, assert_cosine_similarity, get_input_tensors
+from bionemo.llm.model.biobert.lightning import biobert_lightning_module
+from bionemo.testing import megatron_parallel_state_utils
+
+
+def assert_amplify_equivalence(
+    ckpt_path: str,
+    model_tag: str,
+    precision: PrecisionTypes = "fp32",
+    rtol: float | None = None,
+    atol: float | None = None,
+) -> None:
+    tokenizer = BioNeMoAMPLIFYTokenizer()
+
+    input_ids, attention_mask = get_input_tensors(tokenizer)
+    hf_results = load_and_evaluate_hf_amplify(model_tag, precision, input_ids, attention_mask)
+    # gc.collect()
+    # torch.cuda.empty_cache()
+    nemo_results = load_and_evaluate_nemo_amplify(
+        tokenizer,
+        ckpt_path,
+        precision,
+        input_ids,
+        attention_mask,
+    )
+
+    torch.testing.assert_close(hf_results["embeddings"], nemo_results["embeddings"], rtol=rtol, atol=atol)
+    torch.testing.assert_close(hf_results["query_post_rot"], nemo_results["query_post_rot"], rtol=rtol, atol=atol)
+    torch.testing.assert_close(hf_results["key_post_rot"], nemo_results["key_post_rot"], rtol=rtol, atol=atol)
+    torch.testing.assert_close(hf_results["value"], nemo_results["value"], rtol=rtol, atol=atol)
+
+    # torch.testing.assert_close(hf_results["attn_output"], nemo_results["attn_output"], rtol=rtol, atol=atol)
+    assert_cosine_similarity(
+        hf_results["attn_output"],
+        nemo_results["attn_output"],
+        attention_mask.cpu(),
+        rtol=1e-4,
+        atol=1e-4,
+    )
+
+    assert_cosine_similarity(
+        hf_results["attn_linear_output"],
+        nemo_results["attn_linear_output"],
+        attention_mask.cpu(),
+        rtol=1e-4,
+        atol=1e-4,
+    )
+
+    for i, (hf_block_output, nemo_block_output) in enumerate(
+        zip(hf_results["encoder_block_outputs"], nemo_results["encoder_block_outputs"], strict=True)
+    ):
+        assert_cosine_similarity(
+            hf_block_output,
+            nemo_block_output,
+            attention_mask.cpu(),
+            rtol=1e-4,
+            atol=1e-4,
+            msg=f"Encoder block output {i}",
+        )
```

Comment:
So yeah, I have really no idea what to make of this. The accuracy starts degrading steadily through each attention block; this test fails after the second block with an error of ~0.01.

I tried using a non-TE attention block, as well as turning off all the fusions with no luck. Some other things to try would be omitting the mask tokens in the test inputs and see if that helps figure out where this is coming from.

Reply:
Was this fixed by using `DotProductAttention` instead of `TEDotProductAttention` in layer spec?

Reply:
actually no -- it was fixed by me passing the wrong attention mask into the huggingface model 🤦 . 
They raised a warning about it here: https://huggingface.co/chandar-lab/AMPLIFY_120M/commit/e27bb0dff186f9508d13b6ad977c30f8b2e050ed